### PR TITLE
IGNITE-17500 Fix minor issues in RocksDB-based partition storages

### PR DESCRIPTION
**Pull request by sashapolo**
* "Destroy" method returns a future that completes after the data is flushed
* Storage is closed after "destroy" is called
* FlushOptions are cached in RocksDbTableStorage so that they do not get GC-ed

https://issues.apache.org/jira/browse/IGNITE-17500

### CODE REVIEW

File: modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorageTest.java [hunk 1]
```diff
@@ -246,7 +248,8 @@ void storageAdvertisesItIsPersistent() {
         assertThat(storage.isVolatile(), is(false));
     }
 
-    private IgniteBiTuple<TestKey, TestValue> unwrap(BinaryRow binaryRow) {
+    @Nullable
+    private static IgniteBiTuple<TestKey, TestValue> unwrap(@Nullable BinaryRow binaryRow) {
```
Comment (tkalkirill) [inline modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorageTest.java:252]:
```suggestion
    private static @Nullable IgniteBiTuple<TestKey, TestValue> unwrap(@Nullable BinaryRow binaryRow) {
```

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java [hunk 2]
```diff
@@ -193,11 +193,9 @@ public <V> V runConsistently(WriteClosure<V> closure) throws StorageException {
     /** {@inheritDoc} */
     @Override
     public CompletableFuture<Void> flush() {
-        CompletableFuture<Void> flushFuture = new CompletableFuture<>();
-
-        CompletableFuture<Void> oldFuture = flushFuturesByAppliedIndex.put(lastAppliedIndex, flushFuture);
-
-        assert oldFuture == null;
+        CompletableFuture<Void> flushFuture = flushFuturesByAppliedIndex.computeIfAbsent(
```
Comment (tkalkirill) [inline modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java:196]: Why did you remove the assert?

File: modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorageTest.java (hunk 1 above)
Comment (sashapolo) [reply modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorageTest.java:252]: Why?

Comment (tkalkirill) [reply modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorageTest.java:252]: This is just a suggestion: null can be a return value, not a modifier.

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java (hunk 2 above)
Comment (sashapolo) [reply modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java:196]: Because `destroy` could have been called after a Raft command has been applied. This means that we will call `flush` twice: once for the command and once for the destroy, but they will have the same applied index

Comment (tkalkirill) [reply modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java:196]: ok

**Review by tkalkirill — APPROVED** [review]

**Review by SammyVimes — APPROVED** [review]
LGTM
